qiime2/sdk/result.py

```diff
@@ -556,3 +469,132 @@ def get_index_paths(self, relative=True):
     def _repr_html_(self):
         from qiime2.jupyter import make_html
         return make_html(str(self._archiver.path))
+
+
+class ResultCollection:
+    @classmethod
+    def load(cls, directory):
+        """ Determines how to load a Collection of QIIME 2 Artifacts in a
+            directory and dispatches to helpers
+        """
+        if not os.path.isdir(directory):
+            raise ValueError(
+                f"Given filepath '{directory}' is not a directory")
+
+        order_fp = os.path.join(directory, '.order')
+
+        if os.path.isfile(order_fp):
+            collection = cls._load_ordered(directory, order_fp)
+        else:
+            warnings.warn(f"The directory '{directory}' does not contain a "
+                          ".order file. The files will be read into the "
+                          "collection in the order the filesystem provides "
+                          "them in.")
+            collection = cls._load_unordered(directory)
+
+        return collection
+
+    @classmethod
+    def _load_ordered(cls, directory, order_fp):
+        collection = cls()
+
+        with open(order_fp, 'r') as order_fh:
+            for result_name in order_fh.read().splitlines():
+                result_fp = cls._get_result_fp(directory, result_name)
+                collection[result_name] = Result.load(result_fp)
+
+        return collection
+
+    @classmethod
+    def _load_unordered(cls, directory):
+        collection = cls()
+
+        for result in os.listdir(directory):
+            result_fp = os.path.join(directory, result)
+            result_name = result.rstrip('.qza')
+            result_name = result_name.rstrip('.qzv')
+
+            collection[result_name] = Result.load(result_fp)
+
+        return collection
+
+    @classmethod
+    def _get_result_fp(cls, directory, result_name):
+        result_fp = os.path.join(directory, result_name)
+
+        # Check if thing in .order file exists and if not try it with .qza at
+        # the end and if not try it with .qzv at the end
+        if not os.path.isfile(result_fp):
+            result_fp += '.qza'
+
+            if not os.path.isfile(result_fp):
+                result_fp = result_fp.rstrip('.qza')
```

Comment:
rstrip doesn't mean what everyone wants it to mean, and removesuffix doesn't exist yet